_scripts/prune.py

```diff
@@ -0,0 +1,127 @@
+                    res += el_str[1:] if el_str[0] == "\n" else el_str
+                    trim = False
+                else:
+                    res += entity.all
+            file.seek(0)
+            file.write(encode(res, parser.encoding))
+            file.truncate()
+    return len(drop)
+
+
+def prune(branches: list[str]):
+    cwd = getcwd()
+    for branch in branches:
+        if not match(r"[a-z]+[0-9]*", branch):
+            exit(f"Invalid branch names: {branches}")
+
+    removed_data = []
+    removed_files = 0
+    removed_messages = 0
+    refs: dict[str, set[str]] = {}
+    expected = set(branches)
+    for entry in scandir("_data"):
+        branch, ext = splitext(entry.name)
+        if entry.is_file() and ext == ".json":
+            if branch in branches:
+                expected.remove(branch)
+                with open(entry.path, "r") as file:
+                    data: dict[str, list[str]] = json.load(file)
+                for path, keys in data.items():
+                    if path in refs:
+                        refs[path].update(keys)
+                    else:
+                        refs[path] = set(keys)
+            else:
+                print(f"remove {relpath(entry.path, cwd)}")
+                remove(entry.path)
+                removed_data.append(branch)
+    if not refs:
+        exit(f"No data found for: {branches}")
+    if expected:
+        exit(f"Incomplete data! Not found: {expected}")
+
+    for entry in scandir(cwd):
+        if entry.is_dir() and entry.name[0] != "." and entry.name[0] != "_":
+            for root, _, files in walk(entry.path):
+                for file in files:
+                    path = relpath(join(root, file), cwd)
+                    if path not in refs:
+                        print(f"remove {path}")
+                        remove(path)
+                        removed_files += 1
+                    elif refs[path]:
+                        removed_messages += prune_file(path, refs[path])
+    return removed_data, removed_files, removed_messages
+
+
+def write_commit_msg(data: list[str], files: int, messages: int):
+    summary = []
+    for branch in data:
+        summary.append(f"{branch} data")
+    if files:
+        summary.append(f"{files} file" if files == 1 else f"{files} files")
+    if messages:
+        summary.append(
+            f"{messages} message" if messages == 1 else f"{messages} messages"
+        )
+    with open(".prune_msg", "w") as file:
+        file.write(f"Removed: {', '.join(summary)}" if summary else "no changes")
+
+
+if __name__ == "__main__":
+    prog = "python -m _scripts.prune"
+    parser = ArgumentParser(
+        prog=prog,
+        description=description,
+        epilog=f"Example: {prog} master beta release",
+    )
+    parser.add_argument(
+        "branches",
+        nargs="+",
+        help='A list of branch identifiers, e.g. "master beta release".',
+    )
+    args = parser.parse_args()
```

Comment:
We should pass a path like we do for `update.py`.

Reply:
We don't need that here, because we're working in the current directory. In `update.py` we're also working in the current directory, but that script also needs to read from the Firefox checkout, which is in a different directory.